File: 204-count-primes/count-primes.java
class Solution {
    public int countPrimes(int n) {
        // Sieve of Eratosthenes approach
        boolean sqaureOfPrime[] = new boolean[n+1];
        int count = 0;
        for(int i=2; i*i <=n; i++){
            if (sqaureOfPrime[i] == false){
                for(int j=i*i; j<=n; j+=i){
                    sqaureOfPrime[j] = true;
                }
            }
        }
        for(int i=2; i<n; i++){
            if(sqaureOfPrime[i] == false){
                count++;
                }
        }
        return count;

    }
}

File: 204-count-primes/count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        sieve = [False] * (n + 1)
        count = 0

        for i in range(2, int(n ** 0.5) + 1):
            if not sieve[i]:
                for j in range(i * i, n + 1, i):
                    sieve[j] = True

        for i in range(2, n):
            if not sieve[i]:
                count += 1

        return count